### PR TITLE
Base db repository with instance status endpoint

### PR DESCRIPTION
**Pull request by cymon1997**
Implement base db repository 
Added status endpoint to see all instances' status

### CODE REVIEW

File: internal/ping/ping.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package ping
+
+import "context"
+
+type Pingable interface {
```
Comment (cymonevolution) [inline internal/ping/ping.go:5]: I love the idea to generalize all instances that can be pinged for status check

File: pkg/db/client.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+	"github.com/cymon1997/go-template/internal/ping"
+	_ "github.com/go-sql-driver/mysql"
+	_ "github.com/lib/pq"
+	"gorm.io/gorm"
+)
+
+type Client interface {
+	Instance() *gorm.DB
+	ping.Pingable
```
Comment (cymonevolution) [inline pkg/db/client.go:16]: How can this client not implementing `Name()` as it _MUST_ implement `Pingable` interface?

File: pkg/db/client.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		log.Fatal("error init database")
+	}
+	db, err := client.DB()
+	if err != nil {
+		log.Fatal("error connect database")
+	}
+	db.SetMaxOpenConns(cfg.MaxOpenConn)
+	db.SetMaxIdleConns(cfg.MaxIdleConn)
+	db.SetConnMaxIdleTime(time.Duration(cfg.ConnMaxIdleTime) * time.Second)
+	db.SetConnMaxLifetime(time.Duration(cfg.ConnMaxLifeTime) * time.Second)
```
Comment (cymonevolution) [inline pkg/db/client.go:41]: 👍

File: pkg/db/utils.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	case MySQL:
+		return mysql.Open(cfg.DSN)
+	case Postgres:
+		return postgres.Open(cfg.DSN)
+	default:
+		return nil
+	}
+}
+
+func getNowFunc(cfg Config) func() time.Time {
```
Comment (cymonevolution) [inline pkg/db/utils.go:23]: 👍  for this custom timezone

File: pkg/db/client.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+func New(cfg Config) Client {
+	client, err := gorm.Open(getDialector(cfg), &gorm.Config{
+		SkipDefaultTransaction:   false,
+		NowFunc:                  getNowFunc(cfg),
+		PrepareStmt:              true,
+		DisableAutomaticPing:     false,
+		DisableNestedTransaction: false,
+	})
+	if err != nil {
+		log.Fatal("error init database")
```
Comment (cymonevolution) [inline pkg/db/client.go]: Please log the error here

File: pkg/db/timezone.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+		return time.Now().In(loc)
+	}
+)
+
+func InitTimezone(tz string) {
+	syncLoc.Do(func() {
+		var err error
+		loc, err = time.LoadLocation(tz)
+		if err != nil {
+			log.Fatal("error load db timezone")
```
Comment (cymonevolution) [inline pkg/db/timezone.go]: This one also

File: pkg/db/client.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+	}
+	db, err := client.DB()
+	if err != nil {
+		log.Fatal("error connect database")
```
Comment (cymonevolution) [inline pkg/db/client.go]: This one too

**Review by progrevo — CHANGES_REQUESTED** [review]
Great!

File: pkg/db/client.go (hunk 2 above)
Comment (cymon1997) [reply pkg/db/client.go:16]: Seems like the `gorm.DB` already has its own `Name()` implementation @progrevo

File: pkg/db/client.go (hunk 5 above)
Comment (cymon1997) [reply pkg/db/client.go]: Sure

File: pkg/db/client.go (hunk 7 above)
Comment (cymon1997) [reply pkg/db/client.go]: Done

File: pkg/db/timezone.go (hunk 6 above)
Comment (cymon1997) [reply pkg/db/timezone.go]: Done

File: pkg/db/client.go (hunk 2 above)
Comment (cymonevolution) [reply pkg/db/client.go:16]: What's the output gonna be as the result?

Comment (cymon1997) [reply pkg/db/client.go:16]: The output should be depends on the dialector type: 
```
For mysql => mysql
For psql => postgres
```

**Review by progrevo — APPROVED** [review]
Great work!

**Pull request merged**